service-proxy.js

```diff
@@ -112,8 +113,25 @@ function ServiceDispatchHandler(options) {
     self.peersToReap = Object.create(null);
     self.knownPeers = Object.create(null);
 
-    self.reapPeersTimer = null;
-    self.reapPeersPeriod = options.reapPeersPeriod || DEFAULT_REAP_PEERS_PERIOD;
+    self.peerReaper = new IntervalScan(
+        options.reapPeersPeriod || DEFAULT_REAP_PEERS_PERIOD,
+        function reapSinglePeer(hostPort, serviceNames) {
+            self.reapSinglePeer(hostPort, serviceNames);
+        },
+        function getPeersToReap() {
+            var peersToReap = self.peersToReap;
+            self.peersToReap = self.knownPeers;
+            self.knownPeers = Object.create(null);
+            return peersToReap;
+        },
+        self.channel.timers
+    );
+    self.peerReaper.runBeginEvent.on(function onPeerReapBegin(run) {
+        self.logger.info('reaping dead peers', self.extendLogInfo({
+            numPeersToReap: run.keys.length
+        }));
+    });
+    self.peerReaper.start(); // TODO: should IntervalScan be self-starting?
```

Comment:
ServiceProxy itself starts on construction and stops on destruction. Perhaps, having start/stop methods should be a thing you build so you can pause/resume. destroy should stop regardless.